src/Chapter4.hs

```diff
@@ -611,6 +651,15 @@ Implement the 'Monad' instance for our lists.
   maybe a few) to flatten lists of lists to a single list.
 -}
 
+listConcat :: List (List a) -> List a
+listConcat Empty         = Empty
+listConcat (Cons xs xss) = append xs (listConcat xss)
+
+
+instance Monad List where
+  (>>=) :: List a -> (a -> List b) -> List b
+  Empty >>= _ = Empty
+  l     >>= f = listConcat $ fmap f l
```

Comment:
The first pattern with `Empty` is redundant. `fmap` for `Empty` returns `Empty`, and `listConcat` of `Empty` is also `Empty`, so no need to handle separate cases :slightly_smiling_face: Notice, how it's enough to cover such bases cases only in basic fundamental functions, so later they compose nicely and you don't need to worry about missing patterns in such scenarios :earth_africa: